### PR TITLE
Post grid with Custom Post Types

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
#### Functionality:
- [x] Get custom post types from REST API
- [x] Create a store for storing the types 
- [x] Render the custom post types in Block Post 
- [x] PHP function for searching custom types provided by attrs
- [x] Project store instead of local component store

#### Inspector

- [x] Add option to select custom types

### CODE REVIEW

File: src/blocks/posts/edit.js [hunk 1]
```diff
@@ -11,22 +11,67 @@ const {
...
+		getActions() {
+			return actions;
+		},
+		subscribe( listener ) {
+			storeChanged = listener;
+		}
+	};
+};
+
+registerGenericStore( 'otter-posts-block', createPostStore() );
```
Comment (HardeepAsrani) [inline src/blocks/posts/edit.js:63]: Instead of having many stores in each block, we can have one plugin-wide store that we can use.

**Comment by HardeepAsrani** [comment]
>  Add a selection option for the custom types

Not a lot to review until we add that. And make sure user can't select some of the default ones like Dynamic Blocks & Navigation menu.

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel Can you make the picker to a select instead of multi-select checkbox? One is enough.

Also, I can't see the picker if there is no post. And when you change, it takes a lot of time for the picker to update.

**Comment by Soare-Robert-Daniel** [comment]
@HardeepAsrani now it should be better.

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel  It looks good. One thing, make sure the post type names look better, not all capital. And you can get it tested before we merge it.

**Review by HardeepAsrani — APPROVED** [review]

**Comment by HardeepAsrani** [comment]
Looks all good. @irinelenache Can you test from your side? Thank you!

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel One issue that I found was when you switch Block Style from Inspector, it fails.

**Comment by HardeepAsrani** [comment]
Looks to me. You can ask @irinelenache for a final Q/A.

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel I noticed these things happen in the editor:

- [ ] The inline style setting where you choose Grid and List looks like this https://prnt.sc/10solwx
- [ ] More pages get the same featured image when i change the offset to 1 and back to 0 https://www.loom.com/share/1b7a97adec9f4b88894e859b33fcd668 (works fine in front-end)
- [ ] With feedzy plugin installed i get this as an option https://prnt.sc/10sowd4 and if i select it, categories doesn't load in editor https://prnt.sc/10sox0x and this is how they look in front-end https://prnt.sc/10soxxl

This happens with macOS Catalina, both Safari and Chrome

You can write a list with some testing instructions so we better understand how to approach it 🚀

**Comment by HardeepAsrani** [comment]
Thanks @irinelenache - can you fix these @Soare-Robert-Daniel?

**Comment by Soare-Robert-Daniel** [comment]
@irinelenache, can you re-check the issues?

**Comment by HardeepAsrani** [comment]
We really need to merge this to make this release so I hope we can get this one done sooner.

**Comment by irinelenache** [comment]
@HardeepAsrani  @Soare-Robert-Daniel  the issues are fixed, works fine on my end 👍

**Pull request merged**